### ISSUE TITLE
Setting target option does not write any files ...

### ISSUE THREAD

**Issue by eimfach** [addyosmani/critical#439]
I have this options:

```
critical.generate({
  inline: true,
  src: 'index.html',
  width: 1300,
  height: 900,
  extract: true,
  base: './',
  target: {
    css: 'critical.css',
    html: 'index-critical.html',
    uncritical: 'uncritical.css'
  },
  ignore: {atrule: ['@charset']}
}).catch(function(err){
  throw err;
})
```

But it silently does not write any files ...

**Comment by bezoerb** [comment]
which version of critical are you using?

**Comment by eimfach** [comment]
1.3.9

**Comment by bezoerb** [comment]
try installing the 2.x branch with `npm i critical@next`

**Comment by bezoerb** [comment]
the documentation for the `old` version is at https://github.com/addyosmani/critical/tree/1.x

**Comment by eimfach** [comment]
ah I see, okay thanks, will try that

**Comment by eimfach** [comment]
Okay it writes the html file now and I changed target to `target: 'index-critical.html'.
But it does not create a style tag in the header where the critical css should be ...

**Comment by bezoerb** [comment]
hmm... that's weird. Can you create a gist to reproduce?

**Comment by bezoerb** [comment]
@eimfach: Any updates on this?

**Comment by lincolnaleixo** [comment]
not the OP but your solution to update to 2.0 version worked for me. thanks @bezoerb

**Comment by kentr** [comment]
I'm seeing the same thing. In my case I'm using the `html` option and also outputting a css file, which is empty.

My guess is that `critical` is not actually generating the critical `css`, and so there's nothing to put into the `html` file.

**Comment by kentr** [comment]
Yeah, I confirmed that there is no generated `css` by using a callback to log the result.

The `err` parameter to the callback is `null`, so no errors to give clues.

Still debugging...

**Comment by kentr** [comment]
In my case it was due to an old configuration that's no longer supported AFAICT: Passing an `html` string or a local source file that contains links to _external_ stylesheets.

This appears to return "empty" stylesheet documents in [`getStyleseet()`](https://github.com/addyosmani/critical/blob/8c28b57d9e76f2f88e0e6356773fbf9062d84346/src/file.js#L786).

In the debugging, I discovered that remote URLs are supported in the `src` parameter, but they will fail silently if the certificate is self-signed (such as in a dev environment). The `NODE_TLS_REJECT_UNAUTHORIZED` environment variable appears to have no effect on `got`, possibly because `critical` explicitly defaults to [`rejectUnauthorized: true`](https://github.com/addyosmani/critical/blob/8c28b57d9e76f2f88e0e6356773fbf9062d84346/src/file.js#L332).

For me, the solution was to use the actual local URL in `src` instead of a string in `html`, and to set the `request` option with

```javascript
{
  https: {
    rejectUnauthorized: false,
  },
}
```
